File: Testing/Code for Testing/Basic/JS/M004_collections_test.py
from dataclasses import dataclass, field

def __ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class __ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)


def main():
    scores = [95, 87, 72, 100, 63]
    print(scores[0])
    print(scores[4])
    names = ['Alice', 'Bob', 'Carol']
    names.append('Dave')
    print(names[0])
    print(names[3])
    total = 0
    for s in scores:
        total += s
    print(total)
    for name in names:
        print(name)
    return 0


if __name__ == "__main__":
    main()

File: Testing/Code for Testing/Basic/JS/M004_collections_test.js

function main() {
    let scores = [95, 87, 72, 100, 63];
    console.log(scores[0]);
    console.log(scores[4]);
    let names = ["Alice", "Bob", "Carol"];
    names.push("Dave");
    console.log(names[0]);
    console.log(names[3]);
    let total = 0;
    for (const s of scores) {
        total += s;
    }
    console.log(total);
    for (const name of names) {
        console.log(name);
    }
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();
